### PR TITLE
*: support where in admin show ddl jobs statement

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
*: support where in admin show ddl statement

Related parser PR: https://github.com/pingcap/parser/pull/568

eg:
```sql
mysql>admin show ddl jobs where job_type='create table'
+--------+---------+------------+--------------+--------------+-----------+----------+-----------+-----------------------------------+-----------------------------------+--------+
| JOB_ID | DB_NAME | TABLE_NAME | JOB_TYPE     | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | START_TIME                        | END_TIME                          | STATE  |
+--------+---------+------------+--------------+--------------+-----------+----------+-----------+-----------------------------------+-----------------------------------+--------+
| 136    | test    | t          | create table | public       | 1         | 135      | 0         | 2019-09-29 17:41:40.561 +0800 CST | 2019-09-29 17:41:40.565 +0800 CST | synced |
+--------+---------+------------+--------------+--------------+-----------+----------+-----------+-----------------------------------+-----------------------------------+--------+
```

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Release note

 - Support`WHERE` clause in `ADMIN SHOW DDL JOBS` statement

### CODE REVIEW

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -739,9 +739,20 @@ func (b *PlanBuilder) buildAdmin(ctx context.Context, as *ast.AdminStmt) (Plan,
...
 	case ast.AdminShowDDLJobs:
-		p := &ShowDDLJobs{JobNumber: as.JobNumber}
+		p := LogicalShowDDLJobs{JobNumber: as.JobNumber}.Init(b.ctx)
 		p.SetSchema(buildShowDDLJobsFields())
-		ret = p
+		for _, col := range p.schema.Columns {
+			col.UniqueID = b.ctx.GetSessionVars().AllocPlanColumnID()
+		}
+		var np LogicalPlan
+		np = p
```
Comment (bb7133) [inline planner/core/planbuilder.go]: Why not using `ret` directly?

File: planner/core/initialize.go [hunk 2]
```diff
@@ -299,6 +307,14 @@ func (p PhysicalShow) Init(ctx sessionctx.Context) *PhysicalShow {
 	return &p
 }
 
+// Init initializes PhysicalShow.
```
Comment (bb7133) [inline planner/core/initialize.go]:
```suggestion
// Init initializes PhysicalShowDDLJobs.
```

File: planner/core/initialize.go [hunk 3]
```diff
@@ -291,6 +293,12 @@ func (p LogicalShow) Init(ctx sessionctx.Context) *LogicalShow {
 	return &p
 }
 
+// Init initializes LogicalShow.
```
Comment (bb7133) [inline planner/core/initialize.go]:
```suggestion
// Init initializes LogicalShowDDLJobs.
```

File: executor/executor_test.go [hunk 4]
```diff
@@ -456,7 +456,7 @@ func (s *testSuite) TestAdminShowDDLJobs(c *C) {
 	err = tk.Se.CommitTxn(context.Background())
 	c.Assert(err, IsNil)
 
-	re = tk.MustQuery("admin show ddl jobs 1")
```
Comment (lonng) [inline executor/executor_test.go]: Prefer to add some case to `admin show ddl jobs 1 where ...` instead of deleting the former case.

**Pull request merged**